java/vector/src/main/java/org/apache/arrow/vector/file/ArrowWriter.java

```diff
@@ -64,7 +67,7 @@ public void writeRecordBatch(ArrowRecordBatch recordBatch) throws IOException {
   private void checkStarted() throws IOException {
     if (!started) {
       started = true;
-      start();
+      writeMagic();
```

Comment:
There's some ambiguity on whether the file serialization should be the stream bytes exactly, with a magic number header and additional metadata (schema, block index for random access, maybe stats in the future). In this case, the schema would be there twice.

The goal would be to make it as simple as possible (and a minor benefit or being more efficient going from file -> stream: just one seek and read).